pulp_rpm/tests/functional/api/test_publish.py

```diff
@@ -564,3 +563,91 @@ def do_test(self, url=None):
         self.addCleanup(self.distributions.delete, distribution.pulp_href)
 
         return distribution.to_dict()["base_url"]
+
+
+class PublishUnsupportedChecksumTestCase(PulpTestCase):
+    """Test unsupported publish."""
+
+    @classmethod
+    def setUpClass(cls):
+        """Create class-wide variables."""
+        cls.cfg = config.get_config()
+        cls.client = gen_rpm_client()
+        cls.cli_client = cli.Client(cls.cfg)
+        cls.repo_api = RepositoriesRpmApi(cls.client)
+        cls.remote_api = RemotesRpmApi(gen_rpm_client())
+        cls.publications = PublicationsRpmApi(cls.client)
+        cls.distributions = DistributionsRpmApi(cls.client)
+        cls.md5_allowed = "md5" in get_pulp_setting(cls.cli_client, "ALLOWED_CONTENT_CHECKSUMS")
+
+    @skip_if(bool, "md5_allowed", True)
+    def test_publish_with_unsupported_checksum_type(self):
+        """
+        Sync and try publish an RPM repository.
+
+        - Sync repository with on_demand policy
+        - Try to publish with 'md5' checksum type
+        - Publish should fail because 'md5' is not allowed
+
+        This test require disallowed 'MD5' checksum type from ALLOWED_CONTENT_CHECKSUMS settings.
+        """
+        # 1. create repo and remote
+        repo = self.repo_api.create(gen_repo())
+        self.addCleanup(self.repo_api.delete, repo.pulp_href)
+
+        body = gen_rpm_remote(policy="on_demand")
+        remote = self.remote_api.create(body)
+        self.addCleanup(self.remote_api.delete, remote.pulp_href)
+
+        # 2. Sync it
+        repository_sync_data = RpmRepositorySyncURL(remote=remote.pulp_href)
+        sync_response = self.repo_api.sync(repo.pulp_href, repository_sync_data)
+        monitor_task(sync_response.task)
+
+        # 3. Publish and fail
+        publish_data = RpmRpmPublication(repository=repo.pulp_href, package_checksum_type="md5")
+        try:
+            self.publications.create(publish_data)
+        except ApiException as e:
+            if "Checksum must be one of allowed types:" not in e.body:
+                raise ValueError(_("Error message excepted."))
+        else:
+            raise ValueError(_("Publish should have failed."))
```

Comment:
@pavelpicka, could you elaborate how this test (and the next one) works? I 'm not sure I understand how they pass. One way or the other they expect to raise an exception. 
I had an impression that one is supposed to use self.assertRaises. I guess I miss something.

Reply:
@pavelpicka Here's an example of what I mentioned earlier https://github.com/pulp/pulpcore/blob/15e7e587ffe616797cb255a35087c4506a21662e/pulpcore/tests/functional/api/using_plugin/test_tasking.py#L91-L96

Reply:
thank you for example